tests/v2/test_1620-layout-builders.py

```diff
@@ -0,0 +1,168 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import pytest  # noqa: F401
+import numpy as np  # noqa: F401
+import awkward._v2 as ak  # noqa: F401
+
+
+ROOT = pytest.importorskip("ROOT")
+
+
+compiler = ROOT.gInterpreter.Declare
+
+
+def test_data_frame_integers():
+    ak_array_in = ak.Array([1, 2, 3, 4, 5])
+
+    data_frame = ak.to_rdataframe({"x": ak_array_in})
+
+    assert data_frame.GetColumnType("x") == "int64_t"
+
+    ak_array_out = ak.from_rdataframe(
+        data_frame,
+        column="x",
+    )
+    assert ak_array_in.to_list() == ak_array_out["x"].to_list()
+
+
+def test_data_frame_double():
+    ak_array_in = ak.Array([1.1, 2.2, 3.3, 4.4, 5.5])
+
+    data_frame = ak.to_rdataframe({"x": ak_array_in})
+
+    assert data_frame.GetColumnType("x") == "double"
+
+    ak_array_out = ak.from_rdataframe(
+        data_frame,
+        column="x",
+    )
+    assert ak_array_in.to_list() == ak_array_out["x"].to_list()
+
+
+def test_data_frame_char():
+    ak_array_in = ak.Array(["a", "b", "c", "d", "e"])
+
+    data_frame = ak.to_rdataframe({"x": ak_array_in})
+
+    assert data_frame.GetColumnType("x") == "std::string"
+
+    ak_array_out = ak.from_rdataframe(
+        data_frame,
+        column="x",
+    )
+    assert ak_array_in.to_list() == ak_array_out["x"].to_list()
+
+
+def test_data_frame_complex():
+    ak_array_in = ak.Array([1.1 + 0.1j, 2.2 + 0.2j, 3.3 + 0.3j, 4.4 + 0.4j, 5.5 + 0.5j])
+
+    data_frame = ak.to_rdataframe({"x": ak_array_in})
+
+    assert data_frame.GetColumnType("x") == "std::complex<double>"
+
+    ak_array_out = ak.from_rdataframe(
+        data_frame,
+        column="x",
+    )
+    assert ak_array_in.to_list() == ak_array_out["x"].to_list()
+
+
+def test_data_frame_listoffset_integers():
+    ak_array_in = ak.Array([[1], [2, 3, 4], [5]])
+
+    data_frame = ak.to_rdataframe({"x": ak_array_in})
+
+    assert data_frame.GetColumnType("x") == "ROOT::VecOps::RVec<int64_t>"
+
+    ak_array_out = ak.from_rdataframe(
+        data_frame,
+        column="x",
+    )
+    assert ak_array_in.to_list() == ak_array_out["x"].to_list()
+
+
+def test_data_frame_listoffset_listoffset_double():
+    ak_array_in = ak.Array(
+        [
+            [[1.1, 2.2, 3.3]],
+            [[4.4, 5.5]],
+            [[6.6], [7.7, 8.8, 9.9]],
+        ]
+    )
+
+    data_frame = ak.to_rdataframe({"x": ak_array_in})
+
+    # awkward::ListArray_ type
+    # assert data_frame.GetColumnType("x") == "ROOT::VecOps::RVec<double>"
+
+    ak_array_out = ak.from_rdataframe(
+        data_frame,
+        column="x",
+    )
+    assert ak_array_in.to_list() == ak_array_out["x"].to_list()
+
+
+def test_data_frame_vec_of_vec():
```

Comment:
For all of these tests, is the value of `out` correct? They verify that it does not crash, but not that it returns the right value.

Reply:
done